tss2-dlopen/tss2-dlopen-esys.c

```diff
@@ -0,0 +1,2201 @@
+/* SPDX-License-Identifier: BSD-2-Clause */
+/*******************************************************************************
+ * Copyright 2020, Fraunhofer SIT
+ * All rights reserved. *******************************************************************************/
+
+/**
+ * The purpose of this file is to copy it into your project and
+ * include it during compilation if you don't want to link against
+ * libtss2-esys at compile time.
+ * It will attempt to load libtss2-esys.so during runtime.
+ * It will either work similarly to directly linking to libtss2-esys.so
+ * at compile-time or return a NOT_IMPLEMENTED error.
+ *
+ * For new versions of this file, please check:
+ * http://github.com/tpm2-software/tpm2-tss/tss2-dlopen
+*/
+
+#include <dlfcn.h>
+#include <stdio.h>
+#include <tss2/tss2_esys.h>
+
+#define str(s) xstr(s)
+#define xstr(s) #s
```

Comment:
don't get it - why is xstr needed?

Reply:
It's a best practice that has to do with order of macro expansion. You want to expand the contents of s before you stringify it. Don't ask me exactly why, but I learned it somewhere... ;-)

Reply:
ok